### PR TITLE
23

### PR DESCRIPTION
**Pull request by b232wang**

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -0,0 +1,12 @@
+<!DOCTYPE html>
+<html>
+<head>
+  <title>temp</title>
+<script src = "temp.js"></script>
```
Comment (yorkie) [inline index.html]:
``` html
<script src="temp.js"></script>
```

Remove the unnecessary spaces

File: index.html [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+</head>
+
+<body onload = "main()">
```
Comment (yorkie) [inline index.html]: Ditto

File: index.html [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+<canvas id="steven" onclick = "myClick()"  ></canvas>
```
Comment (yorkie) [inline index.html]: Ditto

**Comment by yorkie** [comment]
@b232wang Could you please read https://github.com/weflex/javascript firstly and update your code to follow our coding styles, thank you :-)

File: index.html (hunk 1 above)
Comment (scottoasis) [inline index.html]: @b232wang: Could you use more self-explanatory names like `animalcules.js` than `temp.js`?

File: animalcules.js [hunk 4]
```diff
@@ -0,0 +1,123 @@
+let canvas;
+let ctx;
+let particles = [];
+
+
+function Particle(_x, _y, _vx, _vy,_color,_num){
```
Comment (yorkie) [inline animalcules.js]: could you please remove the _ prefix? thanks

File: animalcules.js [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+    this.x = _x;
+    this.y = _y;
+    this.vx = _vx;
+    this.vy = _vy;
+    this.color = _color;
+    this.num = _num;
+}
+
+function main(){
+    canvas = document.getElementById("canvas");
```
Comment (yorkie) [inline animalcules.js]: single quote

File: animalcules.js [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+    canvas.width = $(window).get(0).innerWidth;
+    canvas.height = $(window).get(0).innerHeight;
```
Comment (yorkie) [inline animalcules.js]: use `window` directly.

**Comment by yorkie** [comment]
Hey Steven, you update doesn't follow our JavaScript coding style https://github.com/weflex/javascript, please read it firstly, thanks.

**Comment by scottoasis** [comment]
@yorkie: As far as I recall, there's an explicit restriction on using Canvas feature here. Correct me if I'm wrong, thanks :-)

**Comment by yorkie** [comment]
I think the canvas restriction might be too hard to a intern?

**Comment by yorkie** [comment]
@b232wang do you have a look at our JavaScript coding style? Your current source code still have much more nits about that, please read it and make changes, thanks.

File: animalcules.js [hunk 7]
```diff
@@ -0,0 +1,130 @@
...
+  // radius
+  const r = Math.random() * 20;
+  // velocity of x
+  const vx = Math.cos(a) * r / 4;
+  // velocity of y
+  const vy = Math.sin(a) * r / 4;
+  let newColor = 0;
+  let myRandom = 0;
+  // random a color
+  while(newColor < 240) {
```
Comment (yorkie) [inline animalcules.js]: white space after the keyword `while`.

File: animalcules.js [hunk 8]
```diff
@@ -0,0 +1,130 @@
...
+    let green = (myRandom >> 8) & 0xff;
+    let blue = myRandom & 0xff;
+    newColor = Math.sqrt(red * red + green * green + blue * blue);
+  }
+  const color = myRandom.toString(16);
+  const p = new Particle(50,50,vx,vy,`#${color}`,1);
+  particles.push(p);
+}
+
+function myClick() {
```
Comment (yorkie) [inline animalcules.js]: Change the function name to be `onClick`.

File: animalcules.js [hunk 9]
```diff
@@ -0,0 +1,130 @@
...
+  const x = event.clientX;
+  const y = event.clientY;
+  particles.every(function(p,index,_ary) {
+    let mx = x - p.x;
+    let my = y - p.y;
+    if(mx >= 0 && mx <= 50 && my >= 0 && my <= 50 ) {
```
Comment (yorkie) [inline animalcules.js]: Ditto

File: animalcules.js [hunk 10]
```diff
@@ -0,0 +1,130 @@
...
+      let tot = p.vx * p.vx + p.vy * p.vy + Math.floor(Math.random() * 4);
+      let nvx = (2 * ((tot | 0)% 2) - 1) * Math.sqrt(Math.floor(Math.random() * tot))
+      let nvy = (2 * (((Math.random() * tot) | 0)% 2) - 1) * Math.sqrt(tot - nvx * nvx);
+      let np = 0;
+      if(p.num * 2 > 1000) {
```
Comment (yorkie) [inline animalcules.js]: Ditto

File: animalcules.js [hunk 11]
```diff
@@ -0,0 +1,130 @@
...
+  }
+  const color = myRandom.toString(16);
+  const p = new Particle(50,50,vx,vy,`#${color}`,1);
+  particles.push(p);
+}
+
+function myClick() {
+  const x = event.clientX;
+  const y = event.clientY;
+  particles.every(function(p,index,_ary) {
```
Comment (yorkie) [inline animalcules.js]: use `.forEach`

File: canvas.css [hunk 12]
```diff
@@ -0,0 +1,3 @@
+* { margin: 0;padding:0; }
+html,body { height:100%, width:100%  }
```
Comment (yorkie) [inline canvas.css]: Should split with semi-colon

File: index.html [hunk 13]
```diff
@@ -0,0 +1,13 @@
+<!DOCTYPE html>
+<html>
+    <head>
+        <title>animalcules</title>
+        <link href="canvas.css"rel="stylesheet" type="text/css">
+        <script type="text/javascript" src="http://ajax.googleapis.com/ajax/libs/jquery/1.8.1/jquery.min.js"></script>
```
Comment (yorkie) [inline index.html]: Please remove the jQuery dependency, your 95% source code are not using this library.

File: animalcules.js [hunk 14]
```diff
@@ -0,0 +1,130 @@
+const $window = $(window)
+let canvas = {};
+let ctx = {};
+let particles = [];
```
Comment (yorkie) [inline animalcules.js]: By the way, could you make a new class like `TheWorld` to save the `particles`, `ctx` and `canvas` so we don't have to need the above global variables, with that we will have a cleaner program structure, thanks.

**Comment by b232wang** [comment]
@yorkie , I did read the JavaScript coding style. Sorry about that I still have some bad code, I will try to fix it as soon as possible. Thanks for your time.

File: animalcules.js [hunk 15]
```diff
@@ -0,0 +1,134 @@
...
+  constructor(contents = []){
+    this.canvas = {};
+    this.ctx = {};
+    this.particles = [];
+  }
+}
+
+const theWorld = new TheWorld();
+
+function Particle(_x, _y, _vx, _vy,_color,_num) {
```
Comment (yorkie) [inline animalcules.js]: Could you please make the `Particle` be  a `class`? Thanks :-)

**Comment by b232wang** [comment]
@yorkie No problem :)

File: animalcules.js [hunk 16]
```diff
@@ -0,0 +1,136 @@
...
+  theWorld.ctx.fillStyle = 'white';
+  theWorld.ctx.fillRect(0,0,theWorld.canvas.width,theWorld.canvas.height);
+  init();
+  setInterval(loop,1000 / 30);
+}
+
+function resizeCanvas() {
+  theWorld.canvas.width = window.innerWidth;
+  theWorld.canvas.height = window.innerHeight;
+};
```
Comment (yorkie) [inline animalcules.js]: This function should be an instance method of `TheWorld`.

File: animalcules.js [hunk 17]
```diff
@@ -0,0 +1,136 @@
...
+  function main() {
+  theWorld.canvas = document.getElementById('canvas');
+  theWorld.canvas.width = window.innerWidth;
+  theWorld.canvas.height = window.innerHeight;
+  theWorld.ctx = theWorld.canvas.getContext('2d');
+  theWorld.ctx.fillStyle = 'white';
+  theWorld.ctx.fillRect(0,0,theWorld.canvas.width,theWorld.canvas.height);
+  init();
+  setInterval(loop,1000 / 30);
+}
```
Comment (yorkie) [inline animalcules.js:102]: Ditto

**Comment by yorkie** [comment]
Please remove all functions and let them be methods of `Particle` or `TheWorld`, thank you :-)

**Comment by b232wang** [comment]
I don't know how to make `main` and `onClick` into any Class. Also, `loop` is inside method `setInterval`, when I try to move loop to a member function, It showed that all function inside `loop` is not defined.

**Comment by yorkie** [comment]
Closed because this has been done :-)

**Pull request closed without merging**